### DISCUSSION TITLE
Make a request on create

### DISCUSSION THREAD

**Discussion by monsterconcursos** [filamentphp/filament#9039] [Help]
Sorry for maybe being a stupid question but I'm a beginner.
How do I perform a specific function when I create a record from a form. Not by submitting when I edit.

Example: Make a request via post to an external API by sending the Filling information in a form field and saving in the database what the API returns.?

**Comment by sinnbeck** [comment]
I assume this is using panels ?

If so you can add this to the create resource
```
protected function handleRecordCreation(array $data): Model
{
    return static::getModel()::create($data);
}
```
https://filamentphp.com/docs/3.x/panels/resources/creating-records#customizing-the-creation-process
